### ISSUE TITLE
可拖拽标签页(Draggable Tabs)

### ISSUE THREAD

**Issue by aiic03** [ant-design/ant-design#9990]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
解决在Electron中使用的时候，让用户能体验到客户端应用中类似Chrome Tabs的效果。

### What does the proposed API look like?
以一个bool开关draggable来设置是否 可拖拽，如：
```
import { Tabs, Button } from 'antd';
const TabPane = Tabs.TabPane;

const operations = <Button>Extra Action</Button>;

ReactDOM.render(
  <Tabs tabBarExtraContent={operations} draggable>
    <TabPane tab="Tab 1" key="1">Content of tab 1</TabPane>
    <TabPane tab="Tab 2" key="2">Content of tab 2</TabPane>
    <TabPane tab="Tab 3" key="3">Content of tab 3</TabPane>
  </Tabs>
, mountNode);
```
实现如图的效果：
![](https://src.works/wp-content/uploads/2017/04/chrome-tabs-demo.gif)

**Comment by imakou** [comment]
I also want this feature as well. Can it be implemented in feature version?

**Comment by abenhamdine** [comment]
perhaps soon, see https://github.com/react-component/tabs/pull/116

**Comment by fenos** [comment]
Is there any current suggestion on how to achieve this?

**Comment by abenhamdine** [comment]
@fenos it's supported, have you read https://github.com/react-component/tabs/pull/116 ?

**Comment by fenos** [comment]
Yes, I commented before jumping on the link! Apologize I'll give it a try this evening! Thanks man

**Comment by fenos** [comment]
@abenhamdine Hi 👋 

I can’t figure out how to make this work. I’ve spent many hours trying to implement draggable tabs with 5-6 different libraries popular and not but I had no luck.

Can you please provide a working example  using any library you’d like to achieve this exact behaviour please. 

It would be amazing!

**Comment by fenos** [comment]
I came to the conclusion that there is a fundamental problem in the library to be able to implement draggable tabs. 

I have a `Draggable` component which I need use as either the wrapper of every DOM element of the following image or as the first element to provide dragging functionalities.

<img width="943" alt="screenshot 2018-12-06 at 09 09 30" src="https://user-images.githubusercontent.com/4754064/49573789-c137a600-f936-11e8-806c-0cd890478629.png">

I'm not able to access to that element from anywhere, so dragging is not possible

**Comment by fenos** [comment]
This is the best library https://github.com/clauderic/react-sortable-hoc to be able to achieve that kind of functionality, but still, I can't figure out how to plug it with this `tab` component.

**Comment by maifine** [comment]
```
 navWrapper={(content) => {
              console.log('content',content);
              return <ReactSortable>{content[0]}</ReactSortable>;
            }}
```
rc-tabs官方给的例子中，因为TabBarTabsNode.js多套了一层div，导致react-sortablejs用不了，不知道有人解决了没？

**Comment by maifine** [comment]
The basic component of tabs, rc-tabs, has the navWrapper() function, but it has problem to use with react-sortablejs, because react-sortablejs will wrap a ‘div’ structure. So I did this by modifying the source of the react-sortablejs.

 ```
<Tabs className={styles.tabNav} onEdit={this.onEdit} activeKey={path}
            navWrapper={(content) => <>
              <ReactSortable>{content[0]}</ReactSortable>
              {content[1]}
            </>}
            onChange={this.onTabChange}
            type="editable-card" hideAdd>……</Tabs>
```
modify the source code of the react-sortablejs:
```
 this.sortable = SortableJS.create(ReactDOM.findDOMNode(this), options);
```
change to:
```
 this.sortable = SortableJS.create(ReactDOM.findDOMNode(this).childNodes[0], options);
```
这样就OK了，主要是感觉sortablejs没有考虑过这种可能有嵌套结构但是又修改不了组件源码的情况，期待antd能够早日把拖拽功能集成到其内部，其实并不复杂。

**Comment by cuijiudai** [comment]
> The basic component of tabs, rc-tabs, has the navWrapper() function, but it has problem to use with react-sortablejs, because react-sortablejs will wrap a ‘div’ structure. So I did this by modifying the source of the react-sortablejs.
> 
> ```
> <Tabs className={styles.tabNav} onEdit={this.onEdit} activeKey={path}
>            navWrapper={(content) => <>
>              <ReactSortable>{content[0]}</ReactSortable>
>              {content[1]}
>            </>}
>            onChange={this.onTabChange}
>            type="editable-card" hideAdd>……</Tabs>
> ```
> 
> modify the source code of the react-sortablejs:
> 
> ```
>  this.sortable = SortableJS.create(ReactDOM.findDOMNode(this), options);
> ```
> 
> change to:
> 
> ```
>  this.sortable = SortableJS.create(ReactDOM.findDOMNode(this).childNodes[0], options);
> ```
> 
> 这样就OK了，主要是感觉sortablejs没有考虑过这种可能有嵌套结构但是又修改不了组件源码的情况，期待antd能够早日把拖拽功能集成到其内部，其实并不复杂。

没有这个navWrapper 属性啊，不起作用
